src/kernels/raw/session/zeromq.node.ts

```diff
@@ -1,51 +1,105 @@
-            traceInfo('ZMQ loaded via fallback mechanism.');
-            sendZMQTelemetry(false, true).catch(noop);
-            return zmq;
+            return loadZmqWithFallbackForWindows(
+                path.join(EXTENSION_ROOT_DIR, 'out', 'node_modules', 'zeromqold'),
+                path.join(EXTENSION_ROOT_DIR, 'out', 'node_modules', 'zeromqold'),
+                true,
+                memento
+            );
         } catch (e2) {
-            sendZMQTelemetry(true, true).catch(noop);
-            traceWarning(`Exception while attempting zmq :`, e.message || e); // No need to display the full stack (when this fails we know why if fails, hence a stack is not useful)
-            traceWarning(`Exception while attempting zmq (fallback) :`, e2.message || e2); // No need to display the full stack (when this fails we know why if fails, hence a stack is not useful)
+            sendZMQTelemetry(true, true, true, memento).catch(noop);
+            traceWarning(`Exception while loading zmq :`, e.message || e); // No need to display the full stack (when this fails we know why if fails, hence a stack is not useful)
+            traceWarning(`Exception while loading zmq (with the fallback) :`, e2.message || e2); // No need to display the full stack (when this fails we know why if fails, hence a stack is not useful)
             throw e2;
         }
     }
 }
 
-async function sendZMQTelemetry(failed: boolean, fallbackTried: boolean = false) {
-    const info = await getDistroInfo().catch(() => <DistroInfo>{ id: '', version_id: '' });
-
-    const telemetryInfo = {
-        ...getPlatformInfo(),
+function loadZmqWithFallbackForWindows(
+    zeromqModuleName: string,
+    moduleDirectory: string,
+    isFallback: boolean,
+    memento: Memento
+) {
+    try {
+        const requireFunc = typeof __webpack_require__ === 'function' ? __non_webpack_require__ : require;
+        const zmq = requireFunc(zeromqModuleName);
+        sendZMQTelemetry(false, isFallback, false, memento).catch(noop);
+        traceInfo(`ZMQ loaded ${isFallback ? 'with' : 'without'} the fallback mechanism.`);
+        return zmq;
+    } catch (e) {
+        traceVerbose(
+            `Failed to load ZMQ ${isFallback ? 'with' : 'without'} the fallback from ${moduleDirectory} due to ${
+                e.message || e
+            }}`
+        );
+        if (os.platform() !== 'win32' || !moduleDirectory) {
+            throw e;
+        }
+        // Special fallback for windows only.
+        // Sometimes the vs runtime dlls do not exist on end user machine.
+        // Try again by copying the files across.
+        const dllsDir = path.join(EXTENSION_ROOT_DIR, 'out', 'node_modules', 'zeromq', 'vcrt');
+        const targetDir = path.join(moduleDirectory, 'prebuilds', 'win32-x64');
+        traceInfo(`Copying dlls from ${dllsDir} to ${targetDir}`);
+        fs.copySync(dllsDir, targetDir, { overwrite: false });
```

Comment:
Sync file IO operations, will be changed to async in debt week, this requires a large amount of refactoring as this function call is expected to be sync and is one of the very first things that happens when the extension loads